src/gui/qgsadvanceddigitizingdockwidget.cpp

```diff
@@ -144,6 +143,21 @@ QgsAdvancedDigitizingDockWidget::QgsAdvancedDigitizingDockWidget( QgsMapCanvas *
   {
     commonAngles << QPair<double, QString>( *it, formatCommonAngleSnapping( *it ) );
   }
+
+  {
+    QAction *action = new QAction( tr( "Snapping to features has priority over common angles" ), mCommonAngleActionsMenu );
+    action->setCheckable( true );
+    mCommonAngleActionsMenu->addAction( action );
+    connect( action, &QAction::changed, this, [ = ]
+    {
+      mSnappingToFeaturesOverridesCommonAngle = action->isChecked();
+      QgsSettings().setValue( QStringLiteral( "/Cad/SnappingToFeaturesOverridesCommonAngle" ), action->isChecked() );
```

Comment:
This should use the new settings framework instead of raw QgsSettings

Reply:
done